### DISCUSSION TITLE
How to clear ScrollArea state from memory?

### DISCUSSION THREAD

**Discussion by BctfN0HUK7Yg** [emilk/egui#1103] [Q&A] [answered]
I using one ScrollArea to show selected content. But ScrollArea store previous state from anoter place. I don't want to keep many states with different Ids. I want to have clear SrollArea each selected content. How to do it?  With TextEdit I use TextEdit::store_state(ui.ctx(),Id::new("text edit"),TextEditState::default()); when change selected content. Is it possible to do so with ScrollArea?

**Comment by emilk** [comment, marked as answer]
If it is good enough to clear the `ScrollArea` _after_ painting it, you could do:

``` rust
let response = ScrollArea::vertical().show(…).response;
ui.ctx().memory().data.remove(response.id);
```
